pkg/xds/generator/ingress_generator.go

```diff
@@ -133,7 +133,7 @@ func tagsFromTargetRef(targetRef common_api.TargetRef) (tags.Tags, bool) {
 		return nil, false
 	}
 
-	return tags.WithTags(mesh_proto.ServiceTag, service), true
+	return mesh_proto.Merge(tags).WithTags(mesh_proto.ServiceTag, service), true
```

Comment:
isn't this iterating through the whole map? seems kind of unnecessary. can we convince your IDE that `WithTags` only reads its caller?

Reply:
Or we can introduce this small optimization: https://github.com/kumahq/kuma/pull/7058

Reply:
Wdyt about https://github.com/kumahq/kuma/pull/7058?

Reply:
Actually, I'm gonna just initialize the Tags map, and it will not use the `Merge` function at all